src/main/java/org/drg/services/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,87 @@
+package org.drg.services.impl;
+
+import org.drg.dtos.requests.UserRequestDto;
+import org.drg.dtos.responses.UserResponseDto;
+import org.drg.exceptions.UserWithSameUsernameException;
+import org.drg.models.UserEntity;
+import org.drg.repositories.UserRepository;
+import org.drg.services.UserService;
+import org.springframework.beans.BeanUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    /*@Autowired
+    private BCryptPasswordEncoder bCryptPasswordEncoder;*/
+
+
+    @Override
+    public Collection<UserResponseDto> findAll() {
+        return userRepository.findAll().stream()
+                .map(user -> UserResponseDto.builder()
```

Comment:
Si te fijas estás repitiendo código en las tres funciones que lo que hacen es transformar la entity en un dto. Para no repetir habría que crear un mapper que dada una entidad user te creara un userdto.

Hay unas librerías muy buenas para hacer este tipo de mappers muy fácilmente, por ejemplo, mapStruct:

https://github.com/mapstruct/mapstruct
https://www.baeldung.com/mapstruct

Si quieres esto me lo apunto yo, para meterle mano, que en el trabajo hago los mappers a mano y no suelo usar estas librerías.

Reply:
apuntado en el "to do" del tablero de trello y asi cerramos esta PR